### ISSUE TITLE
Add ability to update the session/jwt from the server

### ISSUE THREAD

**Issue by camsloanftc** [nextauthjs/next-auth#7342]
### Description 📓

The new `update` function from useSession is good, but it has to be triggered from the client.

It would be great if we could trigger the updates from the server.

I like how [blitzjs handles this](https://blitzjs.com/docs/session-management#change-session-public-data-of-current-user) with a `$setPublicData` function. (It's also nice to have the `$setPrivateData` function as well, but I can live without that.

I think in most cases this would be a more elegant way to update the session, rather than having to authorize and validate the client sessions from the current `update` implementation.

### How to reproduce ☕️

N/a

### Contributing 🙌🏽

Yes, I am willing to help implement this feature in a PR

**Comment by DiegoGonzalezCruz** [comment]
Hi! Do you think that would be useful to update other users jwt?

**Comment by camsloanftc** [comment]
I don’t think you could set another users JWT like this since JWT is stateless, and no info is retained on the server.

If you used database sessions then you could update sessions for other users by a similar technique.

**Comment by khuezy** [comment]
I think this would be useful. I currently hack around this by manually encoding/decoding the `next-auth.session-token` to update it.

**Comment by maccman** [comment]
I definitely think this makes sense. I want to store some redirect params in there.

**Comment by itsbetma** [comment]
I think it can be a great feature, currently, I am implementing a feature that would take advantage of this.

**Comment by knightrun** [comment]
I need a feature to update sessions on the server in my project. Is this feature under development? 
If so, when do you expect it to be completed?
